src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,78 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (clazz.isAnnotationPresent(Component.class)) {
```

Comment:
```suggestion
        if (!clazz.isAnnotationPresent(Component.class)) {
            throw new RuntimeException("Can't initialize instance: " + interfaceClazz.getName() 
            	+ "@Component annotation is absent");
        }
```

`if` конструкції в яких спочатку йде сама умова, а потім десь знизу код, який має виконуватися, якщо умова не дійсна дуже погано читаються, краще одраху викинути помилку, якщо умова не дійсна, а знизу написати код, який має виконуватися якщо все ок

Reply:
Зрозуміло, дякую!